src/scanner.ts

```diff
@@ -392,14 +392,19 @@ export class Scanner {
             }
         }
 
-        return this.source.slice(start, this.index);
+        return {
+            escapedChars: 0,
```

Comment:
I'm afraid this approach will regress scanning performance, due to a object created for every single tokenized identifier.